### PR TITLE
Bind plugin with playground

### LINKED ISSUE #4: Bind playground with the plugin
Bind the playground with the plugin (empty by default). We have to make sure that editing the plugin files refreshes the playground

Also make sure that eslint & prettier works well with the plugin

### PR DESCRIPTION
**Pull request by mdubus**
Closes #4

## What's inside this PR ? 

Binding of the plugin with the playground: 
- change of the plugin entry point
- addition of the plugin in the `gatsby-config` file
- addition of new scripts in package.json and packages to watch files and restart the playground if needed

I also added a test to prove that the binding is made correctly. This will be removed later in the development process.

## How to test ? 

By running the playground, the plugin will programatically create a `Test page` that doesn't exist in the playground, to prove that both are correctly bind together.
To do so, run the following command : 
`yarn playground:dev`

Then go to http://localhost:9000/test , you should see that the test page was successfully created.

To check that the watch mode works perfectly, you can edit the `createPage` inside the `gatsby-node` file, and save. You should see the server restarting and your changes being applied 🎉

### CODE REVIEW

**Comment by bidoubiwa** [comment]
🔥 A few improvements: 

## 1. Testing command

If both build and serve should be done before testing, you can create a script in `package.json` doing both.

Example: 
`yarn playground:build && yarn playground:serve`
 
After some try out it seems that `yarn playground:dev` generates the same result. See point 4.

## 2. Hot reload

> Bind the playground with the plugin (empty by default). We have to make sure that editing the plugin files refreshes the playground

per #4

When updating the plugin we should be able to trigger the rebuild of the playground and then serve the playground again.
Using for example the command of point 1.

Alternatively, hot reload, if `playground:dev` does the same as `playground:build && playground:serve` can be used for hot reloading. 

## 3. Contributing.md

Testing should be added to `### Tests and Linter <!-- omit in toc -->` in `CONTRIBUTING.md`

## 4. Mandatory Build

It seems following your example that building the playground is mandatory to generate the pages.  I tried using `yarn playground:dev` and `yarn playground:build` was not mandatory to create the pages. Is there a reason why you chose to do both instead of dev?

## 5. Issue linking

Which issue does this PR closes?

**Review by bidoubiwa — CHANGES_REQUESTED** [review]
🔥 A few improvements: 

## 1. Testing command

If both build and serve should be done before testing, you can create a script in `package.json` doing both.

Example: 
`yarn playground:build && yarn playground:serve`
 
After some try out it seems that `yarn playground:dev` generates the same result. See point 4.

## 2. Hot reload

> Bind the playground with the plugin (empty by default). We have to make sure that editing the plugin files refreshes the playground

per #4

When updating the plugin we should be able to trigger the rebuild of the playground and then serve the playground again.
Using for example the command of point 1.

Alternatively, hot reload, if `playground:dev` does the same as `playground:build && playground:serve` can be used for hot reloading. 

## 3. Contributing.md

Testing should be added to `### Tests and Linter <!-- omit in toc -->` in `CONTRIBUTING.md`

## 4. Mandatory Build

It seems following your example that building the playground is mandatory to generate the pages.  I tried using `yarn playground:dev` and `yarn playground:build` was not mandatory to create the pages. Is there a reason why you chose to do both instead of dev?

## 5. Issue linking

Which issue does this PR closes?

**Comment by mdubus** [comment]
> ## 4. Mandatory Build
> It seems following your example that building the playground is mandatory to generate the pages. I tried using `yarn playground:dev` and `yarn playground:build` was not mandatory to create the pages. Is there a reason why you chose to do both instead of dev?

You're right, I could have explained how to test it only with the `dev` command instead of the `build + serve`. 
I did so, because this plugin is going to be used on build only. But you're right, it isn't the more convenient way to test for now, therefore I edited the description of this PR in consequence. 
Thanks a lot for your always so constructive feedbacks ! ❤️

File: playground/package.json [hunk 1]
```diff
@@ -8,11 +8,16 @@
...
-    "develop": "gatsby develop",
-    "start": "gatsby develop",
+    "develop": "gatsby develop --open",
+    "start": "gatsby develop --open",
     "build": "gatsby build",
-    "serve": "gatsby serve",
-    "clean": "gatsby clean"
+    "serve": "gatsby serve --open",
+    "clean": "gatsby clean",
+    "restart": "kill-port --port 8000 && yarn start",
```
Comment (bidoubiwa) [inline playground/package.json:16]:
```suggestion
    "restart": "kill-port --port 8000 && yarn develop",
```

File: playground/package.json [hunk 2]
```diff
@@ -8,11 +8,16 @@
     "gatsby"
   ],
   "scripts": {
-    "develop": "gatsby develop",
-    "start": "gatsby develop",
+    "develop": "gatsby develop --open",
+    "start": "gatsby develop --open",
```
Comment (bidoubiwa) [inline playground/package.json:12]:
```suggestion
    "start": "gatsby develop",
```

Could be used for cypress and thus `open` should not be added.

File: package.json [hunk 3]
```diff
@@ -2,11 +2,11 @@
   "name": "gatsby-plugin-meilisearch",
   "version": "0.1.0",
   "description": "Gatsby plugin to index your content to MeiliSearch based on queries",
-  "main": "index.js",
+  "main": "gatsby-node.js",
   "scripts": {
     "lint": "eslint --ext .js .",
     "lint:fix": "eslint --ext .js . --fix",
-    "playground:dev": "yarn --cwd ./playground && yarn --cwd ./playground develop",
+    "playground:dev": "yarn --cwd ./playground && yarn --cwd ./playground watch",
```
Comment (bidoubiwa) [inline package.json:9]:
```suggestion
    "playground:dev": "yarn --cwd ./playground && yarn --cwd ./playground watch",
    "playground:start": "yarn --cwd ./playground && yarn --cwd ./playground start",
```

**Review by bidoubiwa — CHANGES_REQUESTED** [review]

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥

**Comment by mdubus** [comment]
bors merge

**Pull request merged**